src/platform/windows/display_vram.cpp

```diff
@@ -685,6 +636,19 @@ capture_e display_vram_t::snapshot(platf::img_t *img_base, std::chrono::millisec
       BOOST_LOG(error) << "Couldn't query interface [0x"sv << util::hex(status).to_string_view() << ']';
       return capture_e::error;
     }
+
+    // If we don't know the capture format yet, grab it from this texture
+    if(capture_format == DXGI_FORMAT_UNKNOWN) {
+      D3D11_TEXTURE2D_DESC desc;
+      src->GetDesc(&desc);
+      capture_format = desc.Format;
+      BOOST_LOG(info) << "Capture format ["sv << dxgi_format_to_string(capture_format) << ']';
+    }
+  }
+
+  // Now that we know the capture format, we can finish creating the image
+  if(complete_img(img, false)) {
+    return capture_e::error;
```

Comment:
There's a problem with hardware capture on my RX 6600. Using Gravity Mark in Vulkan mode, clicking "Demo" sometimes (but not always) triggers the stream to end with:
```
[2022:12:30:11:12:38]: Error: display_vram_t::complete_img() called with unknown capture format!
```

The failure happen when `snapshot()` is called with format set to `DXGI_FORMAT_UNKNOWN`, but the format doesn't get updated due to `frame_update_flag` being false (which seems to trigger easily with Gravity Mark's Initializing screen presumably due it being low framerate with a static background).

The issue can be resolved by moving the `complete_img()` check into the `frame_update_flag` condition directly above OR changing the error condition to a timeout or OK.

Any of those changes avoids the error, but it brings back a bug that existed before (and was fixed by) your earlier format selection fix. In Gravity Mark, if I move the mouse on the "Initializing" screen, the mouse pointer will have a ghosting effect, perhaps suggesting a format mismatch for the pointer. I don't know precisely how to fix this properly, but if you change the check here to `if (update_flag) {` instead of `if (frame_update_flag) {`, it fixes the pointer ghosting.

Reply:
> I don't know precisely how to fix this properly, but if you change the check here to if (update_flag) { instead of if (frame_update_flag) {, it fixes the pointer ghosting.

This was actually the correct fix. We've been breaking the rules for lifetime of the desktop image we get from the Desktop Duplication API. I'm not sure why this change specifically exposed it (or why my original format change "fixed" it).

The `ReleaseFrame()` docs say:
> After the frame is released, the surface that contains the desktop bitmap becomes invalid; you will not be able to use the surface in a DirectX graphics operation.

Yet we're holding on to that `ID3D11Texture2D` to copy into our buffer when `AcquireNextFrame()` says the desktop image wasn't updated. We depend on the `CopyResource()` call to initialize the destination texture with an updated image before we draw the mouse on top. However, if the old source texture is invalid (which would be the case for a mouse-only update), that copy silently fails (leaving the old contents of the image intact, including the old mouse cursor drawn into it) and we draw the mouse again at the new location.

I pushed a commit to always use the surface we get back from `AcquireNextFrame()` and never hold it beyond the `snapshot()` call.